File: client/src/app/boards/boards.controller.js
export class BoardsController {
  constructor($scope, Board) {
    'ngInject';

    this._Board = Board;
    this._$scope = $scope;

    this.load();
  }

  add(name) {
    new this._Board({name: name}).create().then(result => this.load());
    this.load();
  }

  destroy(board) {
    board.delete().then(result => this.load());
  }

  load() {
    this._Board.query().then(boards => this._$scope.boards = boards);
  }
}

File: app/controllers/boards_controller.rb
class BoardsController < ApplicationController

  def index
    render json: Board.all
  end

  def show
    render json: Board.includes(:columns).find(params[:id]).decorate
  end

  def destroy
    Board.destroy(params[:id])
  end

  def create
    Board.create board_params[:boards]
  end

  private

  def board_params
    params.permit(boards: :name)
  end
end
